### PR TITLE
Make the provisioning api app strict

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Looks good!

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 1]
```diff
@@ -78,7 +79,7 @@ public function __construct(
 	 * @param int $offset
 	 * @return DataResponse
 	 */
-	public function getGroups($search = '', $limit = null, $offset = null) {
+	public function getGroups(string $search = '', $limit = null, $offset = null): DataResponse {
```
Comment (danxuliu) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: Should not `$limit` and `$offset` be declared as integers too?

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -123,7 +124,7 @@ public function __construct($appName,
 	 * @param int $offset
 	 * @return DataResponse
 	 */
-	public function getUsers($search = '', $limit = null, $offset = null) {
+	public function getUsers(string $search = '', $limit = null, $offset = null): DataResponse {
```
Comment (danxuliu) [inline apps/provisioning_api/lib/Controller/UsersController.php]: Same as above (`$limit` and `$offset` declared as integers).

File: apps/provisioning_api/lib/Controller/GroupsController.php (hunk 1 above)
Comment (rullzer) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: Yeah I guess so. But that probably needs more fixing. as I don't want to have ints typehinted that can be `null` etc.

Comment (danxuliu) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: OK, then something for the future ;-)

**Review by danxuliu — APPROVED** [review]
Looks good :+1:

**Pull request merged**